routes/markerSites.js

```diff
@@ -23,15 +23,42 @@ module.exports = function(connection) {
     /* GET all the markers */
     router.get('/markers', function(req, res, next) {
 
-        var query="SELECT * FROM atlas";
+        var query = "SELECT * FROM atlas WHERE";
+
+        //filter by url
+        var criteria = [
+            {
+                param: 'username',
+                col: 'created_by',
+            },
+            {
+                param: 'type',
+                col: 'type',
+            },
+            {
+                param: 'versions',
+                col: 'openmrs_version',
+            },
+            {
+                param: 'dists',
+                col: 'distribution',
+            },
+        ]
+
+        //if parameter exists in the url
+        //push it into params, to use in the sql query
+        var params = [];
+        criteria.forEach(function(crit) {
+            if(req.query[crit.param]) {
+                query += " "+crit.col+"=? AND";
```

Comment:
How about setting `var whereClause = '';` and then using
`whereClause += (params.length === 0 ? ' WHERE ' : ' AND ') + crit.col + "=?";`